File: Ruby/basicMath.rb
# https://ruby-doc.org/docs/ruby-doc-bundle/Tutorial/part_02/user_input.html
# https://www.rubyguides.com/ruby-tutorial/ruby-if-else/

puts "What's your name? "
name = gets.chomp
if name == ""
	name = "Jakob"
end

puts "Type in a number, " + name
inputVar1 = gets.chomp.to_i

puts "Type in another number, " + name
inputVar2 = gets.chomp.to_i

puts "Now add " + inputVar1.to_s + " and " + inputVar2.to_s + " = "
inputSum = gets.chomp.to_i

if inputSum == inputVar1 + inputVar2
	puts "Well done " + name + "!"
else
	puts "Incorrect answer " + name
	puts "Correct answer is: " + (inputVar1 + inputVar2).to_s
end

File: Rust/basicMath.rs
// http://web.mit.edu/rust-lang_v1.25/arch/amd64_ubuntu1404/share/doc/rust/html/book/second-edition/ch02-00-guessing-game-tutorial.html
// Because rust asked so nicely, I decided to rename my variables
// https://doc.rust-lang.org/book/ch03-02-data-types.html
// https://stackoverflow.com/questions/27043268/convert-a-string-to-int

use std::io;
use std::cmp::Ordering;

fn main() {

	println!("What's your name? ");

	let mut input_name = String::new();

	io::stdin().read_line(&mut input_name)
		.expect("Failed to read line");

	//match input_name.cmp(&String) {
	//	Ordering::Equal => let input_name = "Jakob",
	//}

	println!("Type in a number, {}", input_name);

	let mut input_var1 = String::new();

	io::stdin().read_line(&mut input_var1)
		.expect("Failed to read line");

	let input_var1: u32 = input_var1.trim().parse()
		.expect("Please type a number.");

	println!("Type in another number, {}", input_name);

	let mut input_var2 = String::new();

	io::stdin().read_line(&mut input_var2)
		.expect("Failed to read line");

	let input_var2: u32 = input_var2.trim().parse()
		.expect("Please type a number.");

	println!("Now add {} and {}", input_var1, input_var2);

	let mut input_sum = String::new();

	io::stdin().read_line(&mut input_sum)
		.expect("Failed to read line");

	let input_sum: u32 = input_sum.trim().parse()
		.expect("Please type a number.");

	let answer = &input_var1 + &input_var2;

	match input_sum.cmp(&answer) {
		Ordering::Less => println!("Value too low {}", input_name),
		Ordering::Greater => println!("Value too high {}", input_name),
		Ordering::Equal => println!("Well done {}", input_name),
	}
}